include/circt/Dialect/HW/HWTypeInterfaces.td

```diff
@@ -0,0 +1,30 @@
+//===- HWOpInterfaces.td - Operation Interfaces ------------*- tablegen -*-===//
```

Comment:
Nit: This should be `HWTypeInterfaces.td`, and something about Type interfaces not Operation ones.

Mega-nit: I think should probably be "HW type interfaces" or something, i.e. should describe what it is separate from file name, although of course it's very space-limited.